### PR TITLE
Hide hearing worksheet and details links from VSO users

### LINKED ISSUE #8793: Hide Hearing Worksheet link from VSOs
VSOs are using the Case details page just like BVA employees. The Hearing Worksheet lives on the Case Details page and Reader. We've already restricted VSO access so they don't have Reader, but we also want to ensure they can't access the Hearing Worksheet from Case details.

Note: VSOs are only seeing AMA cases right now, and no AMA hearings have been held. This needs to be implemented prior to the first AMA hearing being prepared for, so we should shoot for mid-March at the latest.

## AC
- Hide Hearing Worksheet link from VSOs on Case Details

### PR DESCRIPTION
**Pull request by tomas-nava**
Resolves #8793

### Description
Hides the Hearing Worksheet and Hearing Details links from VSO users.

before:
screen shot 2019-02-15 at 3 01 39 pm

after:
screen shot 2019-02-15 at 2 58 54 pm

### CODE REVIEW

File: spec/feature/queue/case_details_spec.rb [hunk 1]
```diff
@@ -5,8 +5,8 @@
     Timecop.freeze(Time.utc(2020, 1, 1, 19, 0, 0))
   end
 
-  let(:attorney_first_name) { "Robby" }
-  let(:attorney_last_name) { "McDobby" }
+  let(:attorney_first_name) { "Chanel" }
+  let(:attorney_last_name) { "Afshari" }
```
Comment (joeyyang) [inline spec/feature/queue/case_details_spec.rb]: excellent names here

File: spec/feature/queue/case_details_spec.rb [hunk 2]
```diff
@@ -129,12 +133,38 @@
...
+
+        scenario "worksheet and details links are not visible" do
+          visit vso.path
+          click_on "#{appeal.veteran_full_name} (#{appeal.veteran_file_number})"
+
+          expect(page).to have_current_path("/queue/appeals/#{appeal.vacols_id}")
+          scroll_element_in_to_view("#hearings-section")
+          expect(page).to_not have_content("View VLJ Hearing Worksheet")
+          expect(page).to_not have_css("a[href='/hearings/#{hearing.external_id}/worksheet/print?keep_open=true']")
+          expect(page).to_not have_content("View Hearing Details")
```
Comment (joeyyang) [inline spec/feature/queue/case_details_spec.rb]: I'd replace these strings here with their equivalents from `COPY.json`: `COPY::CASE_DETAILS_HEARING_DETAILS_LINK_COPY`

File: spec/feature/queue/case_details_spec.rb [hunk 3]
```diff
@@ -129,12 +133,38 @@
...
+          User.authenticate!(user: vso_user)
+        end
+
+        scenario "worksheet and details links are not visible" do
+          visit vso.path
+          click_on "#{appeal.veteran_full_name} (#{appeal.veteran_file_number})"
+
+          expect(page).to have_current_path("/queue/appeals/#{appeal.vacols_id}")
+          scroll_element_in_to_view("#hearings-section")
+          expect(page).to_not have_content("View VLJ Hearing Worksheet")
```
Comment (joeyyang) [inline spec/feature/queue/case_details_spec.rb]: `COPY::CASE_DETAILS_HEARING_WORKSHEET_LINK_COPY`

File: client/app/queue/CaseHearingsDetail.jsx [hunk 4]
```diff
@@ -66,18 +67,26 @@ class CaseHearingsDetail extends React.PureComponent<Params> {
...
       label: 'Judge',
       value: hearing.heldBy
-    }, {
-      label: '',
-      value: <Link href={`/hearings/${hearing.externalId}/details`}>
-        {COPY.CASE_DETAILS_HEARING_DETAILS_LINK_COPY}
-      </Link>
+    }];
+
+    if (!userIsVsoEmployee) {
```
Comment (joeyyang) [inline client/app/queue/CaseHearingsDetail.jsx]: This seems fine to me, but I think the method would be much more readable if it follows one path if `userIsVsoEmployee` and another path if `!userIsVsoEmployee`, so something like:

```javascript
let hearingAttrs;

if (userIsVsoEmployee) {
  hearingAttrs = [{
     ...
  ]};
} else ( 
  hearingAttrs = [{
    ...
  }];
}
```

Comment (tomas-nava) [reply client/app/queue/CaseHearingsDetail.jsx]: I hear you, but there are multiple conditions being checked to compose this list (the values of `userIsVsoEmployee`, `hearing.disposition`, and `hearing.viewedByJudge` all affect how the "Disposition" element is rendered) which means there are more than two paths.

File: spec/feature/queue/case_details_spec.rb (hunk 3 above)
Comment (tomas-nava) [reply spec/feature/queue/case_details_spec.rb]: fixed with [this commit](https://github.com/department-of-veterans-affairs/caseflow/pull/9463/commits/ea3923d64d9eb1611defc44743fc40273e68acaf).

File: spec/feature/queue/case_details_spec.rb (hunk 2 above)
Comment (tomas-nava) [reply spec/feature/queue/case_details_spec.rb]: fixed with [this commit](https://github.com/department-of-veterans-affairs/caseflow/pull/9463/commits/ea3923d64d9eb1611defc44743fc40273e68acaf).

**Pull request merged**